### ISSUE TITLE
Cache meta information to avoid recalculating them again

### ISSUE THREAD

**Issue by mhewedy** [mhewedy/spwrap#23]
create some static map to cache fullyQualifiedClassName_MethodName and the MetaData object.

Got the idea from https://github.com/marschall/stored-procedure-proxy/wiki/Object-Lifetime (thanks @marschall for exposing such details)

**Comment by marschall** [comment]
You don't even need to create a `fullyQualifiedClassName_MethodName` key, you can simply use the `Method` object as a cache key, see
https://github.com/marschall/stored-procedure-proxy/blob/master/src/main/java/com/github/marschall/storedprocedureproxy/ProcedureCallerFactory.java#L439

**Comment by mhewedy** [comment]
@marschall Thanks for your kind support 👍

**Comment by mhewedy** [comment]
depends on #24

**Comment by mhewedy** [comment]
To do this caching change, first I need to fix #24 because the [getMetaData](https://github.com/mhewedy/spwrap/blob/master/src/main/java/spwrap/proxy/DAOInvocationHandler.java#L31) method doesn't separate between the structure of the stored proc parameters and the return type and between the data that is being passed to the stored procedure for each call.

However it is good to separate between the two aspects, but for these reason I won't do:
1. This will affect the [Param](https://github.com/mhewedy/spwrap/blob/master/src/main/java/spwrap/Caller.java#L302)  that couples the input parameter types with values, which will affect the [Caller interface](https://github.com/mhewedy/spwrap/wiki/Using-Caller-class).

2. There's no actual need for doing the caching as as seen in [refactor_code_log.txt](https://github.com/mhewedy/spwrap/files/816186/refactor_code_log.txt), the avg time to execute getMetaData is almost `0` ms.
